### ISSUE TITLE
Enum in proto file causes pycharm to give spurious errors on pyi

### ISSUE THREAD

**Issue by eNcacz** [dropbox/mypy-protobuf#312]
I have this test proto file:

```protobuf
syntax = "proto3";
package com.example.test;

enum TestEnum {
  ALPHA = 0;
  BETA = 1;
}
```
When I run `protoc --python_out=py --mypy_out=py test.proto`, then I get this pyi file:

```python {highlight=12}
"""
@generated by mypy-protobuf.  Do not edit manually!
isort:skip_file
"""
import builtins
import google.protobuf.descriptor
import google.protobuf.internal.enum_type_wrapper
import typing

DESCRIPTOR: google.protobuf.descriptor.FileDescriptor = ...

class TestEnum(_TestEnum, metaclass=_TestEnumEnumTypeWrapper):
    pass
class _TestEnum:
    V = typing.NewType('V', builtins.int)
class _TestEnumEnumTypeWrapper(google.protobuf.internal.enum_type_wrapper._EnumTypeWrapper[_TestEnum.V], builtins.type):
    DESCRIPTOR: google.protobuf.descriptor.EnumDescriptor = ...
    ALPHA = TestEnum.V(0)
    BETA = TestEnum.V(1)

ALPHA = TestEnum.V(0)
BETA = TestEnum.V(1)
global___TestEnum = TestEnum
```
There are two errors and two warnings which (I think) causes that code completion does not work in PyCharm.

Errors:

* Unresolved reference '_TestEnum': 12
* Unresolved reference '_TestEnumEnumTypeWrapper': 12

Note that classes reference each other so there is no correct order for them in the file.

Warnings:

* Cannot find reference '_EnumTypeWrapper' in 'enum_type_wrapper.pyi':16
* Access to a protected member _EnumTypeWrapper of a module: 16

Expected behavior: Valid pyi file is generated.

**Comment by nipunn1313** [comment]
hiya - appreciate the report!

In a typing context (like a pyi) - it is valid to have such circular typing dependencies (which is why mypy and pyright are happy with such files). Typechecker should run one pass to get symbols and another pass to apply in type context.

That being said, I can potentially see a way to refactor this so that `TestEnum` is last - and `_TestEnumEnumTypeWrapper` references `_TestEnum` rather than `TestEnum` eliminating the circular dependency - which would avoid running into this pycharm bug - so I'll see what I can do. I was able to repro.

As for the warnings - I think it's allowable to access protected members like that.
https://youtrack.jetbrains.com/issue/PY-41243

I think we can add an `__all__` to typeshed so we explicitly export the underscored accesses to avoid it. Another option is to quiet pycharm w/ a comment (`# noinspection PyProtectedMember`) - which might be more desirable because the `_EnumTypeWrapper` is a typing-only construct we created to get stronger value typing.

Thanks for the report! Can work on these.

**Comment by eNcacz** [comment]
Works like a charm. Many thanks!
